charts/templates/configmap-scripts.yaml

```diff
@@ -234,66 +266,59 @@ data:
-    # The third argument is optional and can be used to pass additional options to curl.
-    # Fourth argurment is optional, default is set to true, can be used when custom error handling is required,
-    # if set to true means function will return error and exit if curl fails N_RETRY times
-    # setting to false means function will return response code instead of failing and exiting.
+    # curl_retry_validate(return_error, endpoint, expected_response_code, curl_options...)
     # Retry a curl command until it returns the expected response
     # code or fails N_RETRY times.
     # Use RETRY_INTERVAL to tune the test length.
     # Validate that response code is the same as expected response
     # code or exit with an error.
     #
-    #   $1 :  The target url to test against
-    #   $2 :  The expected response code
-    #   $3 :  Additional options to pass to curl
-    #   $4 :  Option to return error or response code in case of error   
+    #   $1 :  Flag indicating if the script should exit if the given response code is not received ("true" to exit, "false" to return the response code")
+    #   $2 :  The target url to test against
+    #   $3 :  The expected response code
+    #   $4+:  Additional options to pass to curl
     ################################################################
     function curl_retry_validate {
-        local retry_count
-        local return_error="${4:-true}"
+        local retry_count response response_code response_content
+        local return_error=$1; shift
+        local endpoint=$1; shift
+        local expected_response_code=$1; shift
+        local curl_options=("$@")
+
         for ((retry_count = 0; retry_count < N_RETRY; retry_count = retry_count + 1)); do
-            request="curl -m 30 -s -w '%{http_code}' $3 $1"
-            response_code=$(eval "${request}")
-            if [[ ${response_code} -eq $2 ]]; then
-                return "${response_code}"
+            response=$(curl -v -m 30 -w '%{http_code}' "${curl_options[@]}" "$endpoint")
+            response_code=$(tail -n1 <<< "$response")
+            response_content=$(sed '$ d' <<< "$response")
+            if [[ ${response_code} -eq ${expected_response_code} ]]; then
+                return ${response_code}
+            else
+                echo "${response_content}" > /tmp/start-marklogic_curl_retry_validate.log
             fi
+            
             sleep ${RETRY_INTERVAL}
         done
+
         if [[ "${return_error}" = "false" ]] ; then
-            return "${response_code}"  
+            return ${response_code}
         fi
-        error "Expected response code ${2}, got ${response_code} from ${1}." exit
+        [ -f "/tmp/start-marklogic_curl_retry_validate.log" ] && cat start-marklogic_curl_retry_validate.log
+        error "Expected response code ${expected_response_code}, got ${response_code} from ${endpoint}." exit
     }
 
     ################################################################
```

Comment:
We have 2 times the same Function name line 325 and line 373.
But there are doing different steps.
Please review.